src/main/scala/mesosphere/marathon/api/v2/json/GroupUpdate.scala

```diff
@@ -64,14 +64,16 @@ object GroupUpdate {
   }
   def empty(id: PathId): GroupUpdate = GroupUpdate(Some(id))
 
-  implicit val GroupUpdateValidator: Validator[GroupUpdate] = validator[GroupUpdate] { group =>
+  def validNestedGroupUpdateWithBase(base: PathId): Validator[GroupUpdate] = validator[GroupUpdate] { group =>
     group is notNull
 
     group.version is theOnlyDefinedOptionIn(group)
     group.scaleBy is theOnlyDefinedOptionIn(group)
 
     group.id is valid
-    group.apps is valid
-    group.groups is valid
+    group.apps is optional(every(AppDefinition.validNestedAppDefinition(group.id.fold(base)(_.canonicalPath(base)))))
```

Comment:
Unfortunately: since this is a macro, it does not close over external references.
So it has to be part of the validator.